lib/train/transports/ssh_connection.rb

```diff
@@ -316,12 +316,12 @@ def execute_on_channel(cmd)
         channel.exec(cmd) do |_, success|
           abort "Couldn't execute command on SSH." unless success
           channel.on_data do |_, data|
-            yield(data) if block_given?
+            yield(data, channel) if block_given?
```

Comment:
Now we can consume this on the train side with password the data block
```
require 'train'
train = Train.create('ssh', host: 'node1.test', port: 22, user: 'vagrant', key_files: './.vagrant/machines/default/virtualbox/private_key', pty: true)
conn = train.connection

password = "PASSWORD"
res = conn.run_command("su - root -c 'date;'") do |data, ch|
	ch.send_data("#{password}\n") if data.match?("Password:")
end

puts "STDOUT: #{res.stdout}"
puts "STDERR: #{res.stderr}"
```

Reply:
I've tested using the given code and it does work. I also verified that the new block parameter is optional, so any existing code that calls run_command without the `channel` param will not break.

This does put all of the responsibility for handling the password on the calling code.